pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -177,6 +182,19 @@ func (r *ReconcileNodeMaintenance) Reconcile(request reconcile.Request) (reconci
 		return r.reconcileAndError(instance, err)
 	}
 
+	var validQuorum bool
+
+	validQuorum, err = r.checkQuorumValidity(node)
+	if err != nil {
+		reqLogger.Errorf("Can't check quorum validity: %v", err)
+		return r.reconcileAndError(instance, err)
+	}
+	if !validQuorum {
+		err = fmt.Errorf(QuorumViolationErrorMsg)
+		reqLogger.Error(err)
+		return r.reconcileAndError(instance, err)
```

Comment:
I'm not sure that we want to return a reconcile err in this case, it will lead to hot loop, since error means requeue and I don't expect that this will succeed immediately after.

Reply:
fixed, changed to reschedule after five minutes.

Reply:
Doesn't this result in the error being logged twice?

Reply:
reconcileAndError doesn't log the error that is passed as the second argument; at least not in this branch;

Reply:
I thought that the event of detected possible quorum violation  had to be logged with the logger as it is a serious event. Please correct me if I am wrong.